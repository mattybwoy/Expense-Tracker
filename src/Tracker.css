@import url("https://fonts.googleapis.com/css2?family=Press+Start+2P&display=swap");

.Tracker {
  background-color: rgba(128, 128, 128, 0.089);
}

.body {
  background-color: rgba(128, 128, 128, 0.089);
}

footer {
  font-size: 10px;
  font-family: 'Press Start 2P', cursive;
  background-color: #777;
  padding: 10px;
  text-align: center;
  color: white;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
}





