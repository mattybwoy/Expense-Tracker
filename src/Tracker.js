import React, { useState, useEffect } from 'react'
import { Jumbotron } from 'reactstrap'
import logo from './logo.svg'
//import { Button, Container } from 'reactstrap'
//import Form from './components/Form'
import List from './List'
import { Form as BTForm,
FormGroup,
Input,
Label,
Col,
Button,
Container } from 'reactstrap'

const EXPENSE = localStorage.getItem('expenses') ? JSON.parse(localStorage.getItem('expenses')) : []

function Tracker() {
  const [name, setName] = useState('')
  const [amount, setAmount] = useState('')
  const[expenses, setExpenses] = useState(EXPENSE)

  const handleName = event => {
    console.log('Name ', event.target.value)
    setName(event.target.value)
  }

  const handleAmount = event => {
    console.log('Amount ', event.target.value)
    setAmount(event.target.value)
  }

  const handleSubmitForm = event => {
    event.preventDefault();

    if (name!== '' && amount > 0) {
      const expense = { name, amount }
      setExpenses([...expenses, expense])
      console.log(expense)
      setName('')
      setAmount('')
    } else {
      console.log('Invalid expense or amount')
    }
  }

  const handleClearExpenses = () => {
    setExpenses([])
  }

  useEffect(() => {
    localStorage.setItem('expenses', JSON.stringify(expenses))
  }, [expenses])

return (
  <Container>
    <Jumbotron fluid>
      <h2 className= 'display-6' className = 'text-center'>
        Expenses Tracker React App
      <br></br>
        <img src={logo} style={{ width: 50, height: 50 }} alt='react-logo' />
      </h2>
      <div className= 'text-center'>
        <p>
          Total Expenditure: {''}
          <span className='text success'>
            £{''}
            {expenses.reduce((accumulator, currentValue) => {
              return (accumulator += parseFloat(currentValue.amount))
            }, 0)}
          </span>
        </p>
 
      </div>
      <BTForm style={{ margin: 10 }} onSubmit={handleSubmitForm}>
      <FormGroup className='row'>
      <Label for='exampleEmail' sm={2}>
        Name of Expense
      </Label>
      <Col sm={4}>
        <Input
          type='text'
          name='name'
          id='expenseName'
          placeholder='Expense'
          value={name}
          onChange={handleName}
        />
      </Col>
    </FormGroup>
    <FormGroup className='row'>
      <Label for='exampleEmail' sm={2}>
       Amount £
      </Label>
      <Col sm={4}>
        <Input
          type='number'
          name='amount'
          id='expenseAmount'
          placeholder='0.00'
          value={amount}
          onChange={handleAmount}
        />
      </Col>
    </FormGroup>
    <Button type='submit' color='primary' >
      Add
    </Button>{' '}
    <Button type='submit' color ='danger' onClick={handleClearExpenses}>Reset</Button>
  </BTForm>
        <List expenses={expenses} />
    </Jumbotron>
  </Container>
)
}

export default Tracker