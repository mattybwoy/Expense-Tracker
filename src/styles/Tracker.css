@import url("https://fonts.googleapis.com/css2?family=Press+Start+2P&display=swap");

@import url('https://fonts.googleapis.com/css2?family=Castoro&family=Electrolize&family=Noto+Serif+JP:wght@700&display=swap');

.Tracker {
  background-color: rgba(128, 128, 128, 0.089);
}

.text-center {
font-family: 'Castoro', serif;
}

body {
  padding: 50px;
  border-radius: 10px;
  font-family: 'Castoro', serif;
}

footer {
  font-size: 10px;
  font-family: 'Press Start 2P', cursive;
  background-color: #777;
  padding: 10px;
  text-align: center;
  color: white;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
}

Label {
  text-align: right;
}

.add {
  margin-right: 100px;
}

