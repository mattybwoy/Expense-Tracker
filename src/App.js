import React, { useState } from 'react';
import Tracker from './Tracker'

function App() {
  return (
    < Tracker />
  )
}

export default App;