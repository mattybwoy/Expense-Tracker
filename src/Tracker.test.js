import React from 'react';
import '@testing-library/jest-dom'
import { render,screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer';
import App from './App';
import Tracker from './Tracker'

test('renders correctly when there are no expenses', () => {
  const tree = renderer.create( < Tracker / > ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Displays title of app Expenses Tracker React App', () => {
  render(<App />)
  expect(screen.getByText("Expenses Tracker React App")).toBeInTheDocument
})

test('Adds an expense', () => {
  render(<App />)
    const textBox = screen.getByRole('textbox');
    userEvent.type(textBox, 'Coffee')
    const numberBox = screen.getByText('Amount Â£');
    userEvent.click(numberBox, '2')
    userEvent.click(screen.getByText('Add'))
    const note = screen.getByText('Coffee');

    expect(note).toBeInTheDocument();
})