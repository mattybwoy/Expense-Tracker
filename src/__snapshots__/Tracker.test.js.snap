// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no expenses 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron jumbotron-fluid"
  >
    <h2
      className="text-center"
    >
      Expenses Tracker React App
      <br />
      <img
        alt="react-logo"
        src="logo.svg"
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </h2>
    <div
      className="text-center"
    >
      <p>
        Total Expenditure: 
        
        <span
          className="text success"
        >
          £
          
          0
        </span>
      </p>
    </div>
    <form
      className=""
      onSubmit={[Function]}
      style={
        Object {
          "margin": 10,
        }
      }
    >
      <div
        className="row form-group"
      >
        <label
          className="col-sm-2 col-form-label"
          htmlFor="exampleEmail"
        >
          Name of Expense
        </label>
        <div
          className="col-sm-4"
        >
          <input
            className="form-control"
            id="expenseName"
            name="name"
            onChange={[Function]}
            placeholder="Expense"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row form-group"
      >
        <label
          className="col-sm-2 col-form-label"
          htmlFor="exampleEmail"
        >
          Amount £
        </label>
        <div
          className="col-sm-4"
        >
          <input
            className="form-control"
            id="expenseAmount"
            name="amount"
            onChange={[Function]}
            placeholder="0.00"
            type="number"
            value=""
          />
        </div>
      </div>
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="submit"
      >
        Add
      </button>
       
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        type="submit"
      >
        Reset
      </button>
    </form>
    <div>
      <ul
        className="list-group"
      />
    </div>
  </div>
</div>
`;
